Libs/MRML/Core/vtkMRMLPlotNode.h

```diff
@@ -0,0 +1,249 @@
+  // Enum of the available plot types
+  enum {
+    LINE,
+    POINTS,
+    BAR,
+  };
+
+  void PrintSelf(ostream& os, vtkIndent indent);
+
+  //----------------------------------------------------------------
+  /// Standard methods for MRML nodes
+  //----------------------------------------------------------------
+
+  virtual vtkMRMLNode* CreateNodeInstance();
+
+  ///
+  /// Set node attributes
+  virtual void ReadXMLAttributes( const char** atts);
+
+  ///
+  /// Write this node's information to a MRML file in XML format.
+  virtual void WriteXML(ostream& of, int indent);
+
+  ///
+  /// Copy the node's attributes to this object
+  virtual void Copy(vtkMRMLNode *node);
+
+  ///
+  /// Copy the node's attributes to this object but changing the name and type
+  virtual void CopyAndSetNameAndType(vtkMRMLNode *node, const char *name, int Type);
+
+  ///
+  /// Get node XML tag name (like Volume, Model)
+  virtual const char* GetNodeTagName() { return "Plot"; };
+
+  ///
+  /// Method to propagate events generated in mrml
+  virtual void ProcessMRMLEvents(vtkObject *caller, unsigned long event, void *callData);
+
+  ///
+  /// Mark the Table node as references.
+  virtual void SetSceneReferences();
+
+  ///
+  /// Updates this node if it depends on other nodes
+  /// when the node is deleted in the scene.
+  virtual void UpdateReferences();
+
+  ///
+  /// Update the stored reference to another node in the scene.
+  virtual void UpdateReferenceID(const char *oldID, const char *newID);
+
+  //----------------------------------------------------------------
+  /// Get and Set Macros
+  //----------------------------------------------------------------
+
+  ///
+  /// Set and observe a vtkPlot.
+  /// \sa SetInputData()
+  virtual void SetAndObservePlot(vtkPlot* plot);
+
+  ///
+  /// \brief vtkGetObjectMacro
+  /// Get observed plot
+  vtkGetObjectMacro(Plot, vtkPlot);
+
+  ///
+  /// Set and observe Table node ID.
+  /// \sa TableNodeID, GetTableNodeID(), SetInputData()
+  virtual void SetAndObserveTableNodeID(const char *TableNodeID);
+
+  /// Set and observe Table node ID.
+  /// Utility method that conveniently takes a string instead of a char*
+  /// \sa TableNodeID, GetTableNodeID(), SetInputData()
+  virtual void SetAndObserveTableNodeID(const std::string& TableNodeID);
+
+  /// Get Table node ID.
+  /// \sa TableNodeID, SetAndObserveTableNodeID()
+  vtkGetStringMacro(TableNodeID);
+
+  /// Get associated Table MRML node. Search the node into the scene if the node
+  /// hasn't been cached yet. This can be a slow call.
+  /// \sa TableNodeID, SetAndObserveTableNodeID, GetTableNodeID()
+  virtual vtkMRMLTableNode* GetTableNode();
+
+  /// Get the type of the plot (line, scatter, bar)
+  /// \brief vtkGetMacro
+  /// \sa SetTypeAndColor
+  vtkGetMacro(Type, int);
+
+  ///
+  /// Set the type of the plot (line, scatter, bar)
+  virtual void SetType(int type);
+
+  ///
+  /// Get/Set the status of the Plot
+  vtkGetMacro(Dirty, bool);
```

Comment:
Instead of flags like this, all changes should be managed by observations, checking ModifiedTime (change can be detected by comparing ModifiedTime to a stored ModifiedTime), etc.